.ci/jenkins/Jenkinsfile.prod.nightly

```diff
@@ -0,0 +1,199 @@
+                    }
+                }
+            }
+        }
+    }
+    post {
+        failure {
+            emailext body: 'RHBOP ${PRODUCT_VERSION}:nightly-build #${BUILD_NUMBER} was: ' + "${currentBuild.currentResult}" +  '\n' +
+                    'Please look here: ${BUILD_URL} \n' +
+                    ' \n' +
+                    '${BUILD_LOG, maxLines=750}', subject: 'RHBOP ${PRODUCT_VERSION}:nightly-build #${BUILD_NUMBER}: ' + "${currentBuild.currentResult}", to: 'kie-jenkins-builds@redhat.com'
+        }
+        unstable {
+            emailext body: 'RHBOP ${PRODUCT_VERSION}:nightly-build #${BUILD_NUMBER} was: ' + "${currentBuild.currentResult}" +  '\n' +
+                    'Please look here: ${BUILD_URL} \n' +
+                    ' \n' +
+                    'Failed tests: ${BUILD_URL}/testReport \n' +
+                    ' \n' +
+                    '${BUILD_LOG, maxLines=750}', subject: 'RHBOP ${PRODUCT_VERSION}:nightly-build #${BUILD_NUMBER}: ' + "${currentBuild.currentResult}", to: 'kie-jenkins-builds@redhat.com'
+        }
+        fixed {
+            emailext body: 'RHBOP ${PRODUCT_VERSION}:nightly-build #${BUILD_NUMBER} was: ' + "${currentBuild.currentResult}" +  '\n' +
+                    'Please look here: ${BUILD_URL}', subject: 'RHBOP ${PRODUCT_VERSION}:nightly-build #${BUILD_NUMBER}: ' + "${currentBuild.currentResult}", to: 'kie-jenkins-builds@redhat.com'
+        }
+        always {
+            archiveArtifacts artifacts: '**/*.maven.log', fingerprint: false, defaultExcludes: true, caseSensitive: true, allowEmptyArchive: true
+
+            echo 'Generating JUnit report...'
+            junit allowEmptyResults: true, healthScaleFactor: 1.0, testResults: '**/target/*-reports/TEST-*.xml'
+
+            echo 'Archiving logs...'
+            archiveArtifacts excludes: '**/target/checkstyle.log', artifacts: '**/*.maven.log,**/target/*.log', fingerprint: false, defaultExcludes: true, caseSensitive: true, allowEmptyArchive: true
+        }
+        cleanup {
+            cleanWs()
+        }
+    }
+}
+
+def getMessageBody(String optaplannerArchiveUrl, String optaplannerSourcesUrl, String optaplannerQuickstartsSourcesUrl, String optawebSourcesFileUrl, String alreadyBuiltProjects, Map<String, String> versions, Map<String, String> scmHashes) {
+    def alreadyBuiltProjectsArray = (alreadyBuiltProjects ?: '').split(";")
+    return """
+{
+    "version": ${new groovy.json.JsonBuilder(versions).toString()},
+    "scm_hash": ${new groovy.json.JsonBuilder(scmHashes).toString()},
+    "built_projects": ${new groovy.json.JsonBuilder(alreadyBuiltProjectsArray).toString()},
+    "archives": "${optaplannerArchiveUrl}",
+    "sources": {
+        "optaplanner": "${optaplannerSourcesUrl}",
+        "optaplanner-quickstarts": "${optaplannerQuickstartsSourcesUrl}",
+        "optaweb-vehicle-routing": "${optawebSourcesFileUrl}"
+    }
+}"""    
+}
+
+def uploadSources(String project, String folder, Map pmeVariables) {
```

Comment:
don't you think datetimeSuffix is just required and not the whole thing? :thinking:

Reply:
do you mean to have filenames like `optaplanner-quickstarts-sources-2022092909.zip`, i.e., removing the rhbop-8.29.0 prefix?

Reply:
I mean we can probably change method contract to
```
def uploadSources(String project, String folder, String suffix) {
```

Reply:
and then call it by `uploadSources("optaplanner", folder, PME_BUILD_VARIABLES['datetimeSuffix'])`